### PR TITLE
chore(perf): avoid fetching artworks' artists unnecessarily (DIA-988)

### PR DESCRIPTION
**Pull request by joeyAghion**
This **chore** aims to eliminate unnecessary data-loading for artworks' associated `artist` or `artists`, like https://github.com/artsy/force/pull/14858 did for the artwork page's queries.

My process was to search for instances of `artist {` or `artists {`. Any such queries within _artworks_ (the `Artwork` graphql type from Metaphysics) can be resolved shallowly if all of the referenced properties are available in [artists' "short" JSON](https://github.com/artsy/gravity/blob/a5dd184838ab0ed6a6b4f215ce7dedda6546c50e/app/models/domain/artist.rb#L382). Helpfully, relay compilation warns about inconsistent arguments across fragments.

https://artsyproduct.atlassian.net/browse/DIA-988

### CODE REVIEW

File: src/Apps/ArtQuiz/Components/ArtQuizRecommendedArtists.tsx [hunk 1]
```diff
@@ -43,7 +43,7 @@ const ArtQuizRecommendedArtistsFragmentContainer = createFragmentContainer(
         quiz {
           savedArtworks {
             isSaved
-            artist {
+            artist(shallow: true) {
```
[the remark this answers is not in the document]
Comment (joeyAghion) [reply src/Apps/ArtQuiz/Components/ArtQuizRecommendedArtists.tsx:46]: That sounds breaking, unless Metaphysics could short-circuit intelligently based on the referenced properties?

If we were starting from scratch, I imagine the different JSON scopes would simply be different types, `ArtistAbbreviation`, `Artist`, and `ArtistDetail` or something like that.

[the remark this answers is not in the document]
Comment (damassi) [reply src/Apps/ArtQuiz/Components/ArtQuizRecommendedArtists.tsx:46]: Yah, it wouldn't be able to unfortunately. Just wondering if shallow might cover everything in any case in a non-breaking way 👍

File: src/Apps/MyCollection/Routes/EditArtwork/MyCollectionEditArtwork.tsx [hunk 2]
```diff
@@ -136,7 +136,7 @@ export const MyCollectionEditArtworkFragmentContainer = createFragmentContainer(
   {
     artwork: graphql`
       fragment MyCollectionEditArtwork_artwork on Artwork {
-        artist {
+        artist(shallow: true) {
```
Comment (anandaroop) [inline src/Apps/MyCollection/Routes/EditArtwork/MyCollectionEditArtwork.tsx:139]: Wow, this one really demonstrates how _much_ data as accessible from the embedded artist representation. 

**question (non-blocking):** I double-checked this one in my API client to be sure, but I'm curious what the strategy was for verifying these changes?

**Review by anandaroop — APPROVED** [review]
Have to confess, `shallow` hasn't really been on my radar. This is good to know!

File: src/Apps/MyCollection/Routes/EditArtwork/MyCollectionEditArtwork.tsx (hunk 2 above)
Comment (joeyAghion) [reply src/Apps/MyCollection/Routes/EditArtwork/MyCollectionEditArtwork.tsx:139]: It's a multi-step, multi-language process. 😓

First I had to check [how each Metaphysics property is resolved](https://github.com/artsy/metaphysics/blob/main/src/schema/v2/artist/index.ts). Often those are based directly on a corresponding API property, but sometimes they perform additional data-loading or are synthesized based on a few upstream properties. In any case, I'd then have to verify that _all_ of the necessary input properties are available within [artists' short JSON from Gravity](https://github.com/artsy/gravity/blob/a5dd184838ab0ed6a6b4f215ce7dedda6546c50e/app/models/domain/artist.rb#L382). When in doubt I visually confirmed a few locations still rendered the expected data locally, but I didn't check every case that way.

File: src/Apps/ArtQuiz/Components/ArtQuizRecommendedArtists.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (mzikherman) [reply src/Apps/ArtQuiz/Components/ArtQuizRecommendedArtists.tsx:46]: Yea, I like the idea of levering GraphQL types/schema and having that, in addition to `PrivateArtwork`, `AuctionArtwork`, and so forth.

That helps over-fetching b/c we could then do `... on PrivateArtwork` and so forth. Right now for instance - we fetch _everything_ for private/auction/every kind of artwork, and then don't even render any 'private artwork' UI components (or 'auction artwork' UI components, and so forth).

**Pull request merged**